pkg/registry/core/service/storage/rest.go

```diff
@@ -666,6 +666,21 @@ func allocateHealthCheckNodePort(service *api.Service, nodePortOp *portallocator
 func initClusterIP(service *api.Service, allocator ipallocator.Interface) (bool, error) {
 	switch {
 	case service.Spec.ClusterIP == "":
+		// ensure the IP we will allocate is valid within the configured allocator for the cluster
```

Comment:
This needs to be in validation and the strategy needs to be parameterized with the correct settings (single stack, dual stack).